src/snowflake/connector/cursor.py

```diff
@@ -1242,37 +1241,30 @@ def executemany(
 
         return self
 
-    def _result_iterator(
-        self,
-    ) -> Generator[dict, None, None] | Generator[tuple, None, None]:
-        """Yields the elements from _result and raises an exception when appropriate."""
-        try:
-            for _next in self._result:
-                if isinstance(_next, Exception):
-                    Error.errorhandler_wrapper_from_ready_exception(
-                        self._connection,
-                        self,
-                        _next,
-                    )
-                self._rownumber += 1
-                yield _next
-        except TypeError as err:
-            if self._result_state == ResultState.DEFAULT:
-                raise err
-            else:
-                yield None
-
     def fetchone(self) -> dict | tuple | None:
         """Fetches one row."""
         if self._prefetch_hook is not None:
             self._prefetch_hook()
         if self._result is None and self._result_set is not None:
             self._result = iter(self._result_set)
             self._result_state = ResultState.VALID
+
         try:
-            return next(self._result_iterator())
-        except StopIteration:
-            return None
+            _next = next(self._result, None)
```

Comment:
This means `None` will never be a legit result as part of `next(self._result)`. Does this assumption hold? We could also omit the default value of `next` and use `StopIteration` to convey the "end of iteration" message.

Reply:
Right, [None by itself was never a valid value being returned](https://github.com/snowflakedb/snowflake-connector-python/blob/72c4f15dc01299ad319979f5b3294e35876d984f/src/snowflake/connector/result_set.py#L42).